pytest/tests/adversarial/malicious_chain.py

```diff
@@ -0,0 +1,50 @@
+import sys, time
+
+sys.path.append('lib')
+
+valid_blocks_only = False # creating invalid blocks, should be banned instantly
+if "valid_blocks_only" in sys.argv:
+    valid_blocks_only = True # creating valid blocks, should be fixed by doom slug
+
+from cluster import start_cluster
+from adversary import corrupt_node
+from utils import LogTracker
+
+TIMEOUT = 300
+BLOCKS = 25
+MALICIOUS_BLOCKS = 50
+
+nodes = start_cluster(2, 1, 2, None, [["epoch_length", 1000], ["block_producer_kickout_threshold", 80]], {})
+
+started = time.time()
+
+#corrupt_node(nodes[1]) # works if .adv_on_top, not tested otherwise, maybe should be killed and started again
```

Comment:
if you corrupt, you need to kill before corruption and restart after
also, uncomment?

Reply:
uncommenting `corrupt_node` execution